cmd/storage-rest-server.go

```diff
@@ -60,9 +66,22 @@ func (s *storageRESTServer) IsValid(w http.ResponseWriter, r *http.Request) bool
 		s.writeErrorResponse(w, fmt.Errorf("client time %v is too apart with server time %v", requestTime, utcNow))
 		return false
 	}
+	instanceID := r.URL.Query().Get(storageRESTInstanceID)
+	if instanceID != s.instanceID {
+		// Let minio will re-initiate fresh client and validate format.json again
+		s.writeErrorResponse(w, errConnectionStale)
+		return false
+	}
 	return true
 }
 
+// GetInstanceID - returns the instance ID of the server.
+func (s *storageRESTServer) GetInstanceID(w http.ResponseWriter, r *http.Request) {
+	// Skip s.IsValid() check as the request will not have the instanceID
+	w.Header().Set("Content-Length", strconv.Itoa(len(s.instanceID)))
+	w.Write([]byte(s.instanceID))
```

Comment:
We can set this as an HTTP header and be done with it? @krishnasrinivas - no hassle of reading the body etc.